### DISCUSSION TITLE
06. Will the FDA accept data and analyses generated with solutions developed and available as open source?

### DISCUSSION THREAD

**Discussion by MichaelRimler** [phuse-org/OSTCDA#6] [General]
Will the FDA accept data and analyses generated with solutions developed and available as open source?

**Comment by MichaelRimler** [comment]
FDA Statistical Software Clarifying Statement (May 6, 2015)

"FDA does not require use of any specific software for statistical analyses, and statistical software is not explicitly discussed in Title 21 of the Code of Federal Regulations [e.g., in 21CFR part 11]. However, the software package(s) used for statistical analyses should be fully documented in the submission, including version and build identification."

https://www.fda.gov/media/109552/download

Examples of submissions:
- Novo Nordisk: https://www.youtube.com/watch?v=t33dS17QHuA
- Roche: https://posit.co/blog/roche-shifting-to-an-open-source-backbone-in-clinical-trials/
- FDA: https://www.r-consortium.org/using-r-in-regulatory-review

**Comment by rossfarrugia** [comment]
https://rconsortium.github.io/submissions-wg/

**Comment by bms63** [comment]
The FDA does!  but what about all the others?  The R Consortium Submissions WG has only explored FDA.  

Perhaps the title should change to be focused on all Regulatory agencies?

**Reply by rossfarrugia** [reply]
makes sense! we're going to do a webinar with Posit post-summer to share our experience so far submitting in R also including China and EMA (as part of the new data pilot). i heard at least anecdotally that Japan are also open to R.

**Comment by bms63** [comment]
YEah!! I think having any info on submissions to these agencies would be very helpful.  Maybe in a table format.  This is just a quick silly mockup.

| Sponsor/Agency | FDA    | EMA     | PMDA | UK one?  |
| ------------- | ----------- | ------------- | ------------- | ------------- |
| Public Pilots | Submitted | Not done|
| NN | Submitted |
| Roche | Not done|


Several people at Pharmasug were shocked that the FDA took R code after my R-based Submission presentation.  They were under the impression that only SAS code was permitted!!  So we have a lot of work to do!!

**Reply by rossfarrugia** [reply]
yeh, the more real R submission shares from various companies and across different regulators that we get out there the better, and gradually people's fears subside.

p.s. LOL for "UK one?" 😆

**Reply by bms63** [reply]
I couldn't remember its name - UKMHA??

**Reply by rossfarrugia** [reply]
MHRA, close enough though

**Reply by geoskiad** [reply]
This is an excellent point and discussion. Very happy to see it raised here. If not mistaken, in the last R WG for Submissions it was also mentioned that SwissMedic (Health Authority from Switzerland) are interested for greater exposure to R. Based on my experience Health Canada and ANVISA (Brazil) also come often back with questions or have specific rules/requirements related to statistical analysis. But ok. Step by step.

**Reply by bms63** [reply]
@geoskiad you leading Pilot 5 on the R based Submission to SwissMedic?

**Reply by geoskiad** [reply]
> @geoskiad you leading Pilot 5 on the R based Submission to SwissMedic?

No, not at all. Not even a member of it. I am just trying to stay informed of the latest R developments in Pharma world.

**Reply by bms63** [reply]
I was implying you should propose and lead Pilot 5!!

**Comment by KatjaGlassConsulting** [comment]
Thanks to all who participated in the OSTCDA call yesterday. We discussed this topic and would love to get additional input here. Specifically, it would be excellent to **share submission experiences, challenges and links**. @bms63 , @geoskiad and @rossfarrugia - following your suggestions, here is the table we started yesterday and hopefully we get more input.

Sponsor/Agency | FDA | EMA | NMPA | PMDA | Health Canada
-- | -- | -- | -- | -- | --
RConsortium Pilot | Submitted <br/>-	Accepted | Not done
Novo Nordisk <br/>[Presentation](https://www.youtube.com/watch?v=t33dS17QHuA) | Submitted <br/><ul><li>Big data ADaM programmed in R and pre-announced this during Type C and B meeting packages</li><li>Remaining ADaM in SAS and double programmed in R (where applicable)</li><li>All TFL in R</li><li>Following requests from FDA all TFL programs and internal R packages submitted. Packages packed using pkglite.</li><li>[Additional guidance](https://github.com/NNaikp/phuse-css-2023/blob/main/submission-files/instructions-to-programs-in-r.pdf) and informal meeting with FDA statisticians on restoring R environment was held</li></ul> | Submitted content using R | Same as PMDA | <ul><li>Big data ADaM programmed in R and pre-announced during additional eSubmission meeting</li><li>Remaining ADaM in SAS and double programmed in R (where applicable)</li><li>All TFL in R programs submitted</li></ul>| Submitted content using R
Roche <br/>[Presentation](https://www.youtube.com/watch?v=BlJNILSoZlM&t=2187s) | Submitted content using R <br/><ul><li>Initial pushback from the FDA reviewer to accept R, they wanted SAS instead.</li><li>Roche pushed back and submitted in R (based on Technical Conformance Guidance)</li><li>key message - acceptance of R may vary between different agency reviewers</li><li>key message - have the content & format discussions early with the agency</li></ul> | Submitted content using R <br/>(no comments received from the agency so far) | Submitted content using R <br/>(no comments received from the agency so far) |	
SwissMedic
General | | Don’t reanalyse, might be easer |  | Don’t reanalyse, might be easer
Merck | Submitted <br><ul><li>Multiple successful submissions</li><li>GxP platform available</li><li>Reproducible environment</li></ul>			
Astellas Pharma |-	(2021) for using R markdown to submit programs<br/>- [Example](https://www.accessdata.fda.gov/drugsatfda_docs/nda/2022/212608Orig1s006.pdf)

**Reply by bms63** [reply]
This is a very cool table!!  Be great post to put on linkedin asking for other companies inputs

**Reply by vijethamokkapati** [reply]
> Thanks to all who participated in the OSTCDA call yesterday. We discussed this topic and would love to get additional input here. Specifically, it would be excellent to **share submission experiences, challenges and links**. @bms63 , @geoskiad and @rossfarrugia - following your suggestions, here is the table we started yesterday and hopefully we get more input.
> 
> Sponsor/Agency	FDA	EMA	NMPA	PMDA
> RConsortium Pilot	Submitted
> - Accepted	Not done		
> Novo Nordisk	Submitted	- Double programmed with SAS?
> - Didn’t pre announce using R
> - Maybe data preparation in SAS, but TLF in R		
> Roche	Submitted	- Pushed back from reviewers (SAS)
> - Depending on reviewer
> - Content wise no request
> - Asked upfront if it‘s ok to do so	Submitted content from R	Submitted content from R
> SwissMedic				
> General		Don’t reanalyse, might be easer		Don’t reanalyse, might be easer
> Merck	Submitted	- Multiple successful submissions- These were not purely R submissions. R only covered one or more TLFs, so the eCTD package contained both SAS and R content.
> - GxP platform available
> - Reproducible environment		
> Astellas Pharma	- (2021) for using R markdown to submit programs
> - [Example](https://www.accessdata.fda.gov/drugsatfda_docs/nda/2022/212608Orig1s006.pdf)

**Reply by BhupendraMistry** [reply]
The Roche information in the table is not correct. Please amend as follows 

![Phuse R info](https://github.com/user-attachments/assets/393c39ce-ad92-4c23-a9dc-0e4b76796683)

**Reply by KatjaGlassConsulting** [reply]
Thanks, the table for Roche is updated

**Reply by ari-stotle-max** [reply]
For Novo Nordisk it is:

FDA:
- Big data ADaM programmed in R and pre-announced this during Type C and B meeting packages
- Remaining ADaM in SAS and double programmed in R (where applicable)
- All TFL in R
- Following requests from FDA all TFL programs and internal R packages submitted. Packages packed using pkglite.
- [Additional guidance](https://github.com/NNaikp/phuse-css-2023/blob/main/submission-files/instructions-to-programs-in-r.pdf) and informal meeting with FDA statisticians on restoring R environment was held

EMA:
Submitted content using R

Health Canada:
Submitted content using R

PMDA:
- Big data ADaM programmed in R and pre-announced during additional eSubmission meeting
- Remaining ADaM in SAS and double programmed in R (where applicable)
- All TFL in R programs submitted

NMPA:
Same as PMDA

MFDS (Korea):
Same as PMDA

Basically everywhere:
Submitted content using R

**Comment by KatjaGlassConsulting** [comment]
Following our group discussions yesterday (thanks to all who participated) - please continue discussion:

**FDA Experiences**:
-	SAP might mention programming language, e.g. using R -> use what we said that we do
-	How specific and where does the information go with R, e.g. which packages, which versions etc.
-	Who owns reconciliation
	- Sponsor followed what they said they do (using R)
-	Important to point out in early authority meeting to use R
-	ADRG used for documentation, packages, package versions, not executable, but readable
-	Validation of packages has been asked, provided validation records (provided upfront)
-	Fall back SAS option
-	Other authorities seem to be a bit eager for 

**EMA:**

-	Likely follow what is delivered on FDA
-	Pilots running – submission of data, independent of SAS/R
 
**Python:**

-	Interactive dashboards -> exported to figures
-	Less conversation, but it’s within the scope
-	Likely used internally for ML, dashboards similar
-	Data transformations, graph modelling
-	https://github.com/philbowsher/Open-Source-in-New-Drug-Applications-NDAs-FDA 

**Clinical Pharmacology**

-	Using other tools & OS for submissions

**Reply by bms63** [reply]
The FDA reviewers we worked with in the Pilot 3 were okay with ADRG having screenshots for installing `renv` and using the `renv.lock` file to install all package versions from CRAN.  The screenshots helped reduce a lot of verbiage!  Hopefully, opens possibilities for companies to have instructions on installing container infrastructure!

**Comment by KatjaGlassConsulting** [comment]
Another related topic (thanks for all who participated yesterday in the discussions):

**What has been used for OS Submissions:**

-	PDFs for TLF
-	Interactive app
-	Dataset creation in XPT?
-	Pilot4 -> HTML
	- Container
	- webR stuff (quarto)
-	FDA accepts RMD -> did someone so?
-	Specifications for File Format Types Using eCTD Specifications has rmd files for M5 https://www.fda.gov/media/85816/download

**Reply by bms63** [reply]
We should make a push to get `.qmd` files included in the list!

Also - this is very cool blog post coming about being careful when using SAS time versus Unix time epochs and xpts https://github.com/pharmaverse/blog/pull/192

**Comment by MikeKSmith-Pfizer** [comment]
Following discussion at PhUSE EU Connect, some comments captured from the OSTCDA workshop:

- Some regulatory authorities definitely accept R
- Be prepared and learn how to submit
- Be ready to prove that you have validated the software. And be ready for questions (MikeKSmith note: likely questions for clarification about environment, packages, package versions etc.)
- How to have conversations with regulatory authorities?
- Be ready to discuss things like numerical matching.
- Do organisations (sponsor companies) have documentation of why SAS is approved (for regulatory submission)?
